Simplifying README, adding library mode example (#393)
NVIDIA/nv-ingest: README.md

## NVIDIA-Ingest: Multi-modal data extraction

> NVIDIA-Ingest is a scalable, performance-oriented document content and metadata extraction microservice. Including support for parsing PDFs, Word and PowerPoint documents, it uses specialized NVIDIA NIMs (self hosted microservices, or hosted on build.nvidia.com) to find, contextualize, and extract text, tables, charts and images for use in downstream generative applications.

Reviewer: ```suggestion
NVIDIA-Ingest is a scalable, performance-oriented document content and metadata extraction microservice. NV-Ingest includes support for parsing PDFs, Word, and PowerPoint documents. NV-Ingest uses specialized NVIDIA NIMs (self-hosted microservices, or hosted on build.nvidia.com) to find, contextualize, and extract text, tables, charts, and images that you can use in downstream generative applications.
```

### What NVIDIA-Ingest Is ✔️

> A library and microservice that:

Reviewer: ```suggestion
A library and microservice that does the following:
```

- Runs a static pipeline or fixed set of operations on every submitted document.
- Acts as a wrapper for any specific document parsing library.

> For production level performance and scalability, we recommend deploying the pipeline and supporting NIMs via docker-compose or kubernetes (via the provided helm charts).

Reviewer: ```suggestion
For production-level performance and scalability, we recommend that you deploy the pipeline and supporting NIMs by using Docker Compose or Kubernetes. You can use the helm charts provided.
```

Reviewer: Also, can you link from here to the helm charts?

> For hardware and software pre-requisites for container and kubernetes (helm) based deployments, please find [our comprehensive doc site](https://docs.nvidia.com/nv-ingest/user-guide/getting-started/prerequisites/).

Reviewer: ```suggestion
For hardware and software requirements for container- and Kubernetes-based deployments, refer to [Prerequisites](https://docs.nvidia.com/nv-ingest/user-guide/getting-started/prerequisites/).
```

## Library Mode Quickstart

> To facilitate an easier evaluation experience, and for small scale (<100 PDFs) workloads, you can use our "library mode" setup, which depends on NIMs either already self hosted, or, by default, NIMs hosted on build.nvidia.com.

Reviewer: ```suggestion
For small-scale workloads, such as workloads of fewer than 100 PDFs, you can use library mode setup. Library mode set up depends on NIMs that are already self-hosted, or, by default, NIMs that are hosted on build.nvidia.com.
```

To get started using NVIDIA Ingest in library mode, you need to do a few things:
1. Have a [cuDF compatible GPU](https://github.com/rapidsai/cudf?tab=readme-ov-file#cudagpu-requirements)
2. Create a conda environment and install nv-ingest
```
conda create -y --name nvingest python=3.10
conda activate nvingest

conda install -c nvidia -c rapidsai -c rapidsai-nightly nvidia/label/dev::nv_ingest nvidia/label/dev::nv_ingest_client
# conda does not provide milvus packages
pip install opencv-python llama-index-embeddings-nvidia pymilvus 'pymilvus[bulk_writer, model]' milvus-lite
```
3. [Obtain an API key from build.nvidia.com](https://build.nvidia.com/nvidia/llama-3_2-nv-embedqa-1b-v2?snippet_tab=Python&signin=true&api_key=true)
4. Add your key to .env
```.env
NVIDIA_BUILD_API_KEY=...
```

> Note: To use library mode with self hosted NIMs: TODO

Reviewer 2: TODO

> 5. Start ingesting and retrieving documents:

Reviewer 2: Generally seems fine, although I do think having the associated CLI command would be a good idea for anyone who doesn't wants a code-free approach.
